terraform/README.md

```diff
@@ -0,0 +1,154 @@
+
+1. **Install Cloud Foundry CLI tool**
+
+    - On MacOS: `brew install cloudfoundry/tap/cf-cli`
+    - On other platforms: [Download and install cf][cf-install]
+
+1. **Add target environment credentials**
+
+    We are using Terraform to create Cloud Foundry resources in a Cloud.gov account. Creating infrastructure on the Cloud.gov platform requires a Cloud.gov service account username and password. These keys are specific to each cloud.gov organization/space, i.e. each deployment environment, and can be generated by developers who have proper Cloud.gov permissions at any time.
+
+    You can use the same account credentials that are used to deploy applications via CircleCi or follow the steps below to generate a new set of credentials. If you need more information, check out the [service account docs][cloudgov-deployer].
+
+    ```bash
+    # login
+    cf login -a api.fr.cloud.gov --sso
+    # follow temporary authorization code prompts
+    # select org "hhs-acf-ohs-tta", and the space (env) within which you want to build infrastructure
+    # sandbox = ttahub-sandbox
+    # dev = ttahub-dev
+    # staging = ttahub-staging
+    # prod = ttahub-prod
+
+    # create a service instance that can provision service accounts
+    # the value for < YOUR-NAME > can be any version of your name, it isn't significant
+    cf create-service cloud-gov-service-account space-deployer < YOUR-NAME >
+
+    # bind a service key to the service instance
+    cf create-service-key < YOUR-NAME > space-deployer-key
+
+    # return a username/password pair for the service instance
+    cf service-key < YOUR-NAME > space-deployer-key
+    ```
+
+    Add the username and password output from the last command to a `secrets.auto.tfvars` file in each environment directory. Terraform automatically loads this variable definition file. You can also provide variable values via environment variables. For more on this, check out [terraform variable definitions][tf-vars].
+
+    For example, your `terraform/dev/secrets.auto.tfvars` file should look something like this:
+
+    ```
+    cf_user = "some-dev-user"
+    cf_password = "some-dev-password"
+    ```
+
+1. **Create S3 bucket credentials**
+
+    We are using an S3 bucket created by Cloud Foundry in Cloud.gov as our remote backend for Terraform. The backend maintains the "state" of Terraform and makes it possible for multiple developers to implement changes in a linear fashion.
+
+    Follow these directions to create a new service account and generate credentials. If you need more information check out the [services docs][cloudgov-service-keys].
+
+    ```bash
+    # login
+    cf login -a api.fr.cloud.gov --sso
+    # follow temporary authorization code prompts
+    # select org "hhs-acf-ohs-tta", space "infrastructure-config"
+
+    # create a service instance
+    # the value for < YOUR-NAME > can be any version of your name, it isn't significant
+    cf create-service-key ohs-ttahub-iac-state < YOUR-NAME >
```

Comment:
Does this `< YOUR-NAME >` need to be distinct from the `< YOUR-NAME >` in the **Add target environment credentials** step?

Reply:
I don't think it should matter. Remember the service-key here is being created in the `infrastructure-config` space, and is targeting the `ohs-ttahub-iac-state`. The environment cred for dev is created in the `ttahub-dev` space, and is specifically a `space-deployer-key`. So they're two separate _types_ of service keys being created in two separate _spaces_. I can test it out and add direction on that.

Reply:
Gotcha, thanks for explaining this every time I get confused! Hopefully it'll eventually sink in eventually :upside_down_face:

Reply:
Np! This is all going to be difficult to remember because we'll work with terraform infrequently. I tried it out- my hypothesis was correct. I added another line to the comment there to make this clear. :)